definitions/npm/@segment/analytics_v3.11.2/flow_v0.103.x/analytics_v3.11.2.js

```diff
@@ -0,0 +1,189 @@
+  persist?: boolean,
+  ...
+}
+
+declare type SegmentAnalytics$InitOptions = {
+  cookie?: SegmentAnalytics$CookieOptions,
+  metrics?: SegmentAnalytics$MetricsOptions,
+  localStorage?: SegmentAnalytics$StoreOptions,
+  user?: SegmentAnalytics$UserOptions,
+  group?: SegmentAnalytics$GroupOptions,
+  integrations?: {
+    All?: boolean,
+    [integration: string]: boolean | void,
+    ...
+  },
+  ...
+}
+
+declare type SegmentAnalytics$IntegrationsSettings = {
+  [key: string]: any;
+  ...
+}
+
+declare interface SegmentAnalytics$AnalyticsJS {
+  load(writeKey: string, options?: SegmentAnalytics$SegmentOpts): void;
+  identify(
+    userId: string,
+    traits?: Object,
+    options?: SegmentAnalytics$SegmentOpts,
+    callback?: () => void
+  ): void;
+  track(
+    event: string,
+    properties?: Object,
+    options?: SegmentAnalytics$SegmentOpts,
+    callback?: () => void
+  ): void;
+  page(
+    category?: string,
+    name?: string,
+    properties?: Object,
+    options?: SegmentAnalytics$SegmentOpts,
+    callback?: () => void
+  ): void;
+  page(
+    name?: string,
+    properties?: Object,
+    options?: SegmentAnalytics$SegmentOpts,
+    callback?: () => void
+  ): void;
+  group(
+    groupId: string,
+    traits?: Object,
+    options?: SegmentAnalytics$SegmentOpts,
+    callback?: () => void
+  ): void;
+  alias(
+    userId: string,
+    previousId?: string,
+    options?: SegmentAnalytics$SegmentOpts,
+    callback?: () => void
+  ): void;
+  trackLink(
+    elements: JQuery | Element[] | Element,
+    event:
+      | string
+      | {
+          (elm: Element): string,
+          ...
+        },
+    properties?:
+      | Object
+      | {
+          (elm: Element): Object,
+          ...
+        }
+  ): void;
+  trackForm(
+    elements: JQuery | Element[] | Element,
+    event:
+      | string
+      | {
+          (elm: Element): string,
+          ...
+        },
+    properties?:
+      | Object
+      | {
+          (elm: Element): Object,
+          ...
+        }
+  ): void;
+  ready(callback: () => void): void;
+  reset(): void;
+  user(): {
+    id(): string,
+    logout(): void,
+    reset(): void,
+    anonymousId(newId?: string): string,
+    traits(newTraits?: Object): void,
+    ...
+  };
+  group(): {
+    id(): string,
+    traits(newTraits?: Object): void,
+    ...
+  };
+  debug(state?: boolean): void;
+  on(
+    event: string,
+    callback: {
+      (
+        event: string,
+        properties: Object,
+        options: SegmentAnalytics$SegmentOpts
+      ): void,
+      ...
+    }
+  ): void;
+  timeout(milliseconds: number): void;
+}
+
+// Adding type to Window Global as that is how we access Segment functions (via snippet load, see: https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/quickstart/)
+declare type Window = {
+  analytics: SegmentAnalytics$AnalyticsJS,
+  [key: any]: any, // Keep rest of elements untyped (default)
+  ...
+}
+
+declare var window: Window;
```

Comment:
Curious to hear your thoughts on doing this (enforcing the type via Window global), since Segment recommends loading their lib via web snippet (and thus functions are accessible via `window.analytics...` vs. a NPM module

Reply:
You're right about the way it's loaded on window. In our codebase (TS unfortunately), we manually set `analytics` on `Window` like so, so maybe this isn't necessary:

```
interface Window {
    analytics: SegmentAnalytics.AnalyticsJS;
}
```

Reply:
Yeah let's start with that, and if it's an issue for people we can always add it in the definition